### PR TITLE
[12.0][MIG] web_widget_digitized_signature

### PR DESCRIPTION
**Pull request by max3903**
#1048 
Superseeds #1318

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Is the other PR from a member of your team? If not, it sound irespectful to superseed without asking first.

**Comment by max3903** [comment]
@pedrobaeza Yes. He pushed to his own account, instead of our organization.

**Comment by pedrobaeza** [comment]
OK then

**Review by mgosai — APPROVED** [review]

File: web_widget_digitized_signature/__manifest__.py [hunk 1]
```diff
@@ -1,18 +1,20 @@
 # Copyright 2004-2010 OpenERP SA (<http://www.openerp.com>)
 # Copyright 2011-2015 Serpent Consulting Services Pvt. Ltd.
 # Copyright 2017 Tecnativa - Vicent Cubells
+# Copyright (C) 2019 - TODAY, Open Source Integrators
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/__manifest__.py]: Never put `TODAY` reference

File: web_widget_digitized_signature/__manifest__.py [hunk 2]
```diff
@@ -25,4 +27,6 @@
         'static/src/xml/digital_sign.xml',
     ],
     'installable': True,
+    'development_status': 'Stable',
+    'maintainers': ['cubells'],
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/__manifest__.py]: @cubells must accept this

File: web_widget_digitized_signature/__manifest__.py [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+    'version': '12.0.1.0.0',
+    'author': 'Serpent Consulting Services Pvt. Ltd., '
+              'Agile Business Group, '
+              'Tecnativa, '
+              'Open Source Integrators,'
+              'Odoo Community Association (OCA)',
     'website': 'https://github.com/OCA/web',
-    "license": "AGPL-3",
-    "category": 'Web',
+    'license': 'AGPL-3',
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/__manifest__.py:15]: Don't increase the diff if not needed. Disable automatic formatting like `black` for this.

File: web_widget_digitized_signature/__manifest__.py [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-    "author": "Serpent Consulting Services Pvt. Ltd., "
-              "Agile Business Group, "
-              "Tecnativa, "
-              "Odoo Community Association (OCA)",
+    'name': 'Web Widget Digitized Signature',
+    'version': '12.0.1.0.0',
+    'author': 'Serpent Consulting Services Pvt. Ltd., '
+              'Agile Business Group, '
+              'Tecnativa, '
+              'Open Source Integrators,'
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/__manifest__.py]: You are not doing a significant contribution to the module, so you mustn't put in authors. Only on contributors.

File: web_widget_digitized_signature/models/res_users.py [hunk 5]
```diff
@@ -10,10 +10,8 @@ class ResUsers(models.Model):
     _name = 'res.users'
     _inherit = ['res.users', 'mail.thread']
 
-    digital_signature = fields.Binary(
-        string='Signature',
-        oldname="signature_image",
-    )
+    digital_signature = fields.Binary(string='Digital Signature',
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/models/res_users.py:13]: The same about automatic formatting

Comment (pedrobaeza) [reply web_widget_digitized_signature/models/res_users.py:13]: You can add `attachment=True` and add migration script for saving space in DB

File: web_widget_digitized_signature/readme/CONTRIBUTORS.rst [hunk 6]
```diff
@@ -1,4 +1,6 @@
 * Jay Vora <jay.vora@serpentcs.com>
-* Tecnativa <https://www.tecnativa.com>:
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/readme/CONTRIBUTORS.rst]: Please don't plain our formatting system for putting the company URL.

File: web_widget_digitized_signature/readme/CONTRIBUTORS.rst [hunk 7]
```diff
@@ -1,4 +1,6 @@
...
-
-  * Vicent Cubells
+* Lorenzo Battistini <lorenzo.battistini@agilebg.com>
+* Pedro M. Baeza <pedro.baeza@gmail.com>
+* Vicent Cubells <vicent@vcubells.net>
+* Mayank Gosai <mgosai@opensourceintegrators.com>
+* Maxime Chambreuil <mchambreuil@opensourceintegrators.com>
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/readme/CONTRIBUTORS.rst:6]: 2 contributors for this minimum changes?

File: web_widget_digitized_signature/static/src/js/digital_sign.js [hunk 8]
```diff
@@ -22,8 +22,8 @@ odoo.define('web_widget_digitized_signature.web_digital_sign', function(require)
         placeholder: "/web/static/src/img/placeholder.png",
         init: function() {
             this._super.apply(this, arguments);
-            this.$('> img').remove();
-            this.$('.signature > canvas').remove();
+//            this.$('> img').remove();
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/static/src/js/digital_sign.js]: You must not comments things: or remove them, or keep it, but also why are you removing them?

File: web_widget_digitized_signature/views/res_users_view.xml [hunk 9]
```diff
@@ -1,4 +1,3 @@
-<?xml version="1.0" ?>
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/views/res_users_view.xml]: Don't remove this. You are breaking XML format

File: web_widget_digitized_signature/views/web_digital_sign_view.xml [hunk 10]
```diff
@@ -1,9 +1,11 @@
-<?xml version="1.0" encoding="UTF-8"?>
 <odoo>
 
-    <template id="web_widget_digitized_signature_backend" name="web_widget_digitized_signature assets" inherit_id="web.assets_backend">
+    <template id="web_widget_digitized_signature_backend"
```
Comment (pedrobaeza) [inline web_widget_digitized_signature/views/web_digital_sign_view.xml]: No automatic syntax.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Adding a lot of noise, lobbying and no valuable changes: the module would work in 12.0 as is.

**Comment by max3903** [comment]
@mgosai Can you take care of the comments please to get this merged? Thanks.

File: web_widget_digitized_signature/__manifest__.py (hunk 2 above)
Comment (max3903) [reply web_widget_digitized_signature/__manifest__.py]: @pedrobaeza If he doesn't, what do you suggest? He wrote the module, right?

File: web_widget_digitized_signature/__manifest__.py (hunk 3 above)
Comment (max3903) [reply web_widget_digitized_signature/__manifest__.py:15]: @pedrobaeza I think migration is a great opportunity to clean up those inconsistencies of having simple and double quote within the same manifest file.

File: web_widget_digitized_signature/views/web_digital_sign_view.xml (hunk 10 above)
Comment (max3903) [reply web_widget_digitized_signature/views/web_digital_sign_view.xml]: @pedrobaeza Same thing. Migration is a good opportunity to make this file easily reviewable.

File: web_widget_digitized_signature/__manifest__.py (hunk 3 above)
Comment (pedrobaeza) [reply web_widget_digitized_signature/__manifest__.py:15]: Well, I don't see them as inconsistencies. Both are allowed in Python the same and it doesn't produce any visual problem. We don't have any guideline that forces to unify them. Even if so, that must be done on a separate commit `[REF] style` or similar.

File: web_widget_digitized_signature/__manifest__.py (hunk 2 above)
Comment (pedrobaeza) [reply web_widget_digitized_signature/__manifest__.py]: Well, as said, he must accept this charge

File: web_widget_digitized_signature/views/web_digital_sign_view.xml (hunk 10 above)
Comment (pedrobaeza) [reply web_widget_digitized_signature/views/web_digital_sign_view.xml]: The same answers: I don't see this as an improvement and any guideline says about that, and also it mustn't be included in the migration commit.

**Comment by pedrobaeza** [comment]
Please squash all these commits

**Comment by nikul-serpentcs** [comment]
@max3903 @mgosai Can you please [Travis ](https://travis-ci.org/OCA/web/jobs/556887195#L593-L596)

**Comment by max3903** [comment]
@nikul-serpentcs Done!

**Comment by pedrobaeza** [comment]
This still has undesired auto-format changes I have already mentioned.

**Review by nikul-serpentcs — APPROVED** [review]
Only Code Review LGTM :+1: 

@max3903 Please check/modify  others reviewer comment's also

**Comment by max3903** [comment]
@pedrobaeza All the changes are desired.

**Comment by pedrobaeza** [comment]
No, they aren't desired, and if you want to do it, you must split that changes in a separate commit apart from the migration as stated.

Please don't disobey next time as you always do and launch the merge command without agreement.

**Comment by rafaelbn** [comment]
Hi @max3903 ,

I don't think is good practice to merge with a disagree of a PCS, you should then ask maybe another PSC in  @OCA/web-maintainers but not merge.

You add Coyright, Author and 2 contribution by changin " by ' https://github.com/OCA/web/pull/1325/commits/70080385593de26142320187e23e96217059e364 plus 4 lines of code https://github.com/OCA/web/pull/1325/commits/70080385593de26142320187e23e96217059e364#diff-7a2fc2224315edb7571db98b9462f6e1

We must also maintain the same criteria for all when it comes to attributing contributions or authorizations. If you make a migration of a module with a diff of 40 lines of which code change are 4 lines, then what is the criterion. You have to be fair.

This is not something that of course takes away the dream, it is not harmful for Tecnativa. But it is harmful and unfair in my personal opinion for OCA and the rest of the contributors.

Regards

**Comment by max3903** [comment]
Hi @rafaelbn,

The changes I pushed are cosmetic changes to facilitate code readibility and have consistency:

* Having double quote and single quote in the same manifest doesn't make sense to me.
* Having XML file with 120+ characters line either.

I don't see the point of doing separate pull requests of the same module for that and asking people to review this separately. Let's not waste everyone else time on this. Especially when we ask people later to squash commits...

Also this is a Tecnativa module that we decided to maintain, so give me a break. If you wanted to take care of it and maintain it, you had plenty of time to provide an alternative maintainer.

I think we discussed it long enough last year, this kind of behavior is negative for the community. We should welcome contributions and improve them, not reject them.

**Comment by rafaelbn** [comment]
> We should welcome contributions and improve them, not reject them

Absolutely agree here, I'm very concerned about expanding OCA

**Comment by pedrobaeza** [comment]
I was asking for 2 separated commits, not 2 PRs as now it's very difficult to know which changes are due to 12.0 needs and what are only cosmetic changes. It's not too much to ask.

**Comment by hbrunn** [comment]
I agree it's a good thing to ask contributors to keep the history clean. So very often I'm stuck in some combined commit when I want to follow how some code came to place, I appreciate every effort to be strict there.

But given the 2 PRs / 2 commits misunderstanding, can we file it as such and move on? And let's learn from it and write some suggestions about this in `CONTRIBUTING.rst`.

**Pull request merged**